pinot-core/src/test/java/org/apache/pinot/queries/InterSegmentAggregationMultiValueQueriesTest.java

```diff
@@ -595,7 +595,7 @@ public void testFilteredAggregations() {
         new DataSchema.ColumnDataType[]{DataSchema.ColumnDataType.LONG});
     ResultTable expectedResultTable =
         new ResultTable(expectedDataSchema, Collections.singletonList(new Object[]{370236L}));
-    QueriesTestUtils.testInterSegmentsResult(brokerResponse, 370236L, 0L, 0L, 400000L, expectedResultTable);
+    QueriesTestUtils.testInterSegmentsResult(brokerResponse, 370236L, 400000L, 0L, 400000L, expectedResultTable);
```

Comment:
Is this expected?  I believe this is number of docs scanned in filter.  Any idea why there's now more docs scanned in the filter than before? Has index usage changed in the tests?

Reply:
This is expected because we do scan this number of docs to resolve `column3 > 0` (verified with query without the aggregated filter). The reason why it shows as 0 before is because we don't track the scans for main filter in `CombinedFilterOperator` (there is a TODO in `BlockDocIdSet.toNonScanDocIdSet()`). With the change, we won't create the `CombinedFilterOperator`, thus the scans can be correctly reported.